### PR TITLE
Return default value to prevent missing return warning.

### PR DESCRIPTION
**Pull request by iwanders**
This addresses #84.

Original suggested fix was to both return the default value and add a `ROS_BREAK()`, however the latter would require modifying the package dependencies for code is never reached. This fix returns the default value of `T` to silence warnings when compiling with clang, since this part of the code is never reached the behaviour is identical.

### CODE REVIEW

**Review by tfoote — APPROVED** [review]
Thanks for the patch. I've added a comment. I'll wait for CI before merging.

**Comment by iwanders** [comment]
@tfoote , can you please merge this?

**Pull request merged**
